app/uk/gov/hmrc/customs/declaration/connectors/MdgWcoDeclarationConnector.scala

```diff
@@ -62,11 +65,22 @@ class MdgWcoDeclarationConnector @Inject()(http: HttpClient,
     http.POSTString[HttpResponse](url, xml.toString())
       .recoverWith {
         case httpError: HttpException => Future.failed(new RuntimeException(httpError))
-      }
-      .recoverWith {
         case e: Throwable =>
           logger.error(s"Call to wco declaration submission failed. url=$url")
           Future.failed(e)
       }
   }
+
+  override protected def circuitBreakerConfig: CircuitBreakerConfig =
+    CircuitBreakerConfig(
+      serviceName = configKey,
+      numberOfCallsToTriggerStateChange = config.declarationsCircuitBreakerConfig.numberOfCallsToTriggerStateChange,
+      unavailablePeriodDuration = config.declarationsCircuitBreakerConfig.unavailablePeriodDurationInMillis,
+      unstablePeriodDuration = config.declarationsCircuitBreakerConfig.unstablePeriodDurationInMillis
+    )
+
+  override protected def breakOnException(t: Throwable): Boolean = t match {
+    case _: BadRequestException | _: NotFoundException | _: Upstream4xxResponse => false
```

Comment:
I think it might be better to return true in Upstream5xxResponse and false on everything else

Reply:
Most other HMRC implementations favour this set of exceptions so I'll leave this unaltered.